python/src/iceberg/schema.py

```diff
@@ -134,7 +138,7 @@ def _lazy_id_to_accessor(self) -> dict[int, Accessor]:
 
     def as_struct(self) -> StructType:
         """Returns the underlying struct"""
-        return self._struct
+        return StructType(*self.fields)
```

Comment:
I think I would opt to keep an inner struct like this did before, and then define `fields` to return `this.struct.fields`

Reply:
I tried this, but that creates a lot of awkward code since it serializes as fields. I can cache it on the first call, wdyt?

Reply:
No worries, we can keep it as is for now.